controller/backup_controller.go

```diff
@@ -499,6 +499,31 @@ func (bc *BackupController) reconcile(backupName string) (err error) {
 		return nil
 	}
 
+	defer func() {
+		// If the backup is in the final state, delete the snapshot if needed
+		if _, ok := backupInfo.Labels[types.RecurringJobLabel]; ok {
+			// leave the snapshot management for recurring jobs to the `SettingNameAutoCleanupRecurringJobBackupSnapshot` setting.
+			return
+		}
+
+		cleanupSnap, err := bc.ds.GetSettingAsBool(types.SettingNameAutoCleanupSnapshotAfterOnDemandBackupCompleted)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to get the setting %v", types.SettingNameAutoCleanupSnapshotAfterOnDemandBackupCompleted)
+			return
+		}
+		if cleanupSnap && backup.Spec.SnapshotName != "" {
+			errDel := bc.ds.DeleteSnapshot(backup.Spec.SnapshotName)
+			if errDel != nil {
+				if apierrors.IsNotFound(errDel) {
+					return
+				}
+				err = errDel
+				log.WithError(err).Error("Failed to delete the snapshot")
+				return
+			}
+		}
```

Comment:
how about change to the following structure, more readable
```
			if errDel := bc.ds.DeleteSnapshot(backup.Spec.SnapshotName); errDel != nil && apierrors.IsNotFound(errDel)  {
				err = errDel
				log.WithError(err).Error("Failed to delete the snapshot")
				return
			}
```

Reply:
Sure. More simple. Modified. Thanks.